api/src/test/java/uk/gov/dhsc/htbhf/claimant/message/processor/DetermineEntitlementMessageProcessorTest.java

```diff
@@ -146,16 +224,162 @@ void shouldUpdateClaimWhenClaimantIsNotEligible(EligibilityStatus eligibilitySta
+                currentCycleChildrenDobs);
+        given(messageContextLoader.loadDetermineEntitlementContext(any())).willReturn(context);
+        given(pregnancyEntitlementCalculator.isEntitledToVoucher(any(), any())).willReturn(false);
+        //Had children in last cycle who are under 4 at the start of the current payment cycle.
+        given(childDateOfBirthCalculator.hadChildrenUnder4AtStartOfPaymentCycle(any())).willReturn(true);
+        //Children in previous cycle are now be over 4.
+        given(childDateOfBirthCalculator.hadChildrenUnderFourAtGivenDate(any(), any())).willReturn(false);
+
+        //Eligibility - any status that isn't ELIGIBLE has the same effect here.
+        EligibilityAndEntitlementDecision decision = aDecisionWithStatusAndChildren(INELIGIBLE, qualifyingBenefitEligibilityStatus, currentCycleChildrenDobs);
+        given(eligibilityAndEntitlementService.evaluateExistingClaimant(any(), any(), any())).willReturn(decision);
+
+        //Current payment cycle voucher entitlement mocking
+        Message message = aValidMessageWithType(DETERMINE_ENTITLEMENT);
+
+        //When
+        MessageStatus messageStatus = processor.processMessage(message);
+
+        //Then
+        runCommonVerifications(context, decision, message, messageStatus, ClaimStatus.EXPIRED);
+        verify(childDateOfBirthCalculator).hadChildrenUnder4AtStartOfPaymentCycle(context.getPreviousPaymentCycle());
+        verify(pregnancyEntitlementCalculator).isEntitledToVoucher(NOT_PREGNANT, context.getCurrentPaymentCycle().getCycleStartDate());
+        verifyZeroInteractions(determineEntitlementNotificationHandler);
+    }
+
+    @ParameterizedTest(name = "Qualifying benefit status={0}")
+    @ValueSource(strings = {"CONFIRMED", "NOT_CONFIRMED"})
+    void shouldExpireClaimWhenClaimantWasPregnantButNoLongerWithNoChildrenInPreviousCycle(
+            QualifyingBenefitEligibilityStatus qualifyingBenefitEligibilityStatus) {
+        //Given
+        List<LocalDate> previousCycleChildrenDobs = NO_CHILDREN;
+        List<LocalDate> currentCycleChildrenDobs = NO_CHILDREN;
+        LocalDate previousCycleExpectedDeliveryDate = EXPECTED_DELIVERY_DATE_IN_TWO_MONTHS;
+        LocalDate currentCycleExpectedDeliveryDate = NOT_PREGNANT;
+        DetermineEntitlementMessageContext context = buildMessageContext(
+                previousCycleExpectedDeliveryDate,
+                currentCycleExpectedDeliveryDate,
+                previousCycleChildrenDobs,
+                currentCycleChildrenDobs);
+        given(messageContextLoader.loadDetermineEntitlementContext(any())).willReturn(context);
+        LocalDate currentPaymentCycleStartDate = context.getCurrentPaymentCycle().getCycleStartDate();
+        LocalDate previousPaymentCycleStartDate = context.getPreviousPaymentCycle().getCycleStartDate();
+        lenient().when(pregnancyEntitlementCalculator.isEntitledToVoucher(NOT_PREGNANT, currentPaymentCycleStartDate)).thenReturn(false);
```

Comment:
Why is lenient needed here?

Reply:
Because this an the next line are calling the same method on the same method twice, if I remove lenient then Mockito instantly fails the test.